playwright/src/test/java/com/microsoft/playwright/TestPageScreenshot.java

```diff
@@ -146,6 +146,17 @@ void maskShouldWork() {
     assertThrows(AssertionFailedError.class, () -> assertArrayEquals(screenshot, originalScreenshot));
   }
 
+  @Test
+  void maskShouldWorkForLocator() {
```

Comment:
note: We should have migrated more mask tests from [upstream](https://github.com/microsoft/playwright/blob/0467c28722044a4ae84aac055f3f1c6059a0bf70/tests/page/page-screenshot.spec.ts#L459-L485) while adding the feature, it would help to avoid bugs like this one.